src/main/groovy/grails/plugins/mail/graph/reader/GraphEmailReaderService.groovy

```
@@ -118,20 +150,33 @@ class GraphEmailReaderService {
     */
 
     MailFolder createMailFolder(GraphConfig graphConfig, String mailFolderName) {
-        log.debug("Creating mail folder ${mailFolderName} for config ${graphConfig.configName}")
-        MailFolder mailFolder = new MailFolder(displayName: mailFolderName, isHidden: false)
-        GraphServiceClient serviceClient = graphApiClient.getClientFor(graphConfig)
-        return serviceClient.me()
-                .mailFolders()
-                .post(mailFolder)
+        log.info("[GRAPH_EMAIL] [CREATE_MAIL_FOLDER] [STARTED] - EMAIL_ADDRESS=${graphConfig.emailAddress} | FOLDER_NAME=${mailFolderName}")
+
+        try {
+            MailFolder mailFolder = new MailFolder(displayName: mailFolderName, isHidden: false)
+            GraphServiceClient serviceClient = graphApiClient.getClientFor(graphConfig)
+            MailFolder created = serviceClient.me().mailFolders().post(mailFolder)
+            log.info("[GRAPH_EMAIL] [CREATE_MAIL_FOLDER] [SUCCESS] - EMAIL_ADDRESS=${graphConfig.emailAddress} | FOLDER_NAME=${mailFolderName}")
+            return created
+        } catch (Exception ex) {
```

@RxL-Farhan-Ali we should not catch all exceptions or catch specific only.

---

updated removed try catch